HIPs/incubation/web3j.md

```diff
@@ -0,0 +1,204 @@
+
+# Status
+
+Proposal
+
+# Solution
+
+Web3j is made up of a number of components. It was designed to be as accessible as possible to developers, and the design of these components reflects this.
+
+In order to bring the greatest benefit to its users, the expectation is these components collectively would make up the Hyperledger Web3j project.
+
+These components are as follows:
+
+- The core Web3j library provides the Ethereum integration capabilities to support use cases that utilise the JSON-RPC protocol and smart contracts
+- Smart contract testing components that allow users to develop and test Solidity smart contracts within their JVM project workflow
+- Build plugins that integrate the Solidity compiler and Solidity library dependencies within the JVM project
+- A command line interface (CLI) to simplify working with Web3j
+
+These are discussed in further detail below.
+
+## Core Web3j library
+
+The [core Web3j library](https://github.com/web3j/web3j) was designed to be highly modular, so that users could easily incorporate the minimal parts of the library for their projects. This consideration was especially important in servicing the Android community.
+
+The notable modules include:
+
+- **rlp:** [Recursive Length Prefix (RLP)](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/) serialisation
+- **crypto:** cryptographic operations and key management 
+- **abi:** [Contract Application Binary Interface (ABI)](https://docs.soliditylang.org/en/latest/abi-spec.html) encoding and decoding
+- core: core Web3j components including the JSON-RPC implementation and abstractions for wallets
+- utils: utility functions including type conversions
+- **besu, geth, eea: Ethereum** client specific functionality
+- codegen: code generators for creating Java type wrappers for Solidity smart contracts
+- contracts: smart contract specific types
+
+## Smart Contract Testing
+
+The Web3j core library is complemented by the following projects that facilitate unit and integration testing of Solidity smart contracts.
+
+The [Web3j EVM](https://github.com/web3j/web3j-evm) project provides an embedded EVM that allows developers to write applications that simulate transactions taking place on the EVM. 
+
+The EVM contained within the project is the Hyperledger Besu EVM. This ensures that the EVM can be run within the local JVM without the need to orchestrate a full Ethereum node via Docker or similar tools.
+
+One of the significant advantages of using the Hyperledger Besu EVM is that developers work with a production-grade EVM during the development process. In other non-JVM development environments, it's unlikely that developers will have at their disposal an embedded EVM that uses the same code as a widely used Ethereum client.
+
+The Web3j EVM project is complemented by the [Web3j Unit](https://github.com/web3j/web3j-unit) project. This project is used to inject different EVM environments into unit and integration tests.
```

Comment:
Would this repository also be coming to Hyperledger?

Reply:
Yes, as comment above.